### ISSUE TITLE
could not create a temp file

### ISSUE THREAD

**Issue by PromontoryProtean** [webprofusion/certify#413]
I am running into an issue on Windows Server 2019. When I open Certify a warning window pops up that says:
Could not create a temp file (). Windows has a limit of 65535 files in the temp folder (C:\Users\(my-username-here)\AppData\Local\Temp\2\). Clear temp files before proceeding. The directory name is invalid.

When I attempt to browse to that directory it does not exist. I have attached a screenshot of the error message for more clarification.

After I click through the warning I have been able to use Certify normally, so I am not sure to what extent an actual problem exists here.
Thanks,
John

![certify-error](https://user-images.githubusercontent.com/10098842/50350018-a86cea00-050b-11e9-9e30-6861d6960c19.jpg)

**Comment by webprofusion-chrisc** [comment]
Hi John, it may be that windows is reporting the wrong temp location. Where does it go if you open %temp% in windows explorer?

The message is there because some operations like updating the database etc create temp files. If we can't do that then we probably can't operate normally and other apps on the server may have similar problems.

It could be that the reported temp folder is just not accessible, or it could be that your actual temp folder has the max number of file/directory objects.

**Comment by webprofusion-chrisc** [comment]
Check this SO answer for the list of env variables that set your temp path. Its worth having this working for the stability of your server in general: https://stackoverflow.com/questions/39127373/tpath-gettempfilename-replies-the-directory-name-is-invalid/39127920

**Comment by PromontoryProtean** [comment]
Thanks for the tip. When I tried to browse to %temp% the location did not exist. It was returning C:\Users(my-username-here)\AppData\Local\Temp\2 which seems like a weird path to me. This is a brand new Windows Server 2019 VM install on Azure so I have no idea how that setting could have gotten tweaked.

Good news is I gave the server a reboot and then the temp path started working. I'll have to do some digging to see what the default temp path should be and how it could have gotten changed. I've had some weird problems with Server 2019 on Azure so far, so I wouldn't be surprised if this is another bug in the OS. I would have stuck with Server 2016, but I was desperate to get away from the "let's install a cumulative update and then force you to wait over an hour while your production server is offline and pray that it comes back" problem.

Thanks,
John

**Comment by PromontoryProtean** [comment]
I did some digging on this and the problem is related to connecting to the server using remote desktop. There is a group policy setting called "Do not use temporary folders per session" that can force all sessions to use the same temp location. If this setting is NOT configured (which is the default), when connecting via remote desktop the temp folder location will end in something with a \2 or \3 and so on.

I just ran into the problem again when interacting with a session that had been connected for a couple of days - the temp folder just disappeared. I logged off and logged back on and the problem went away. When I browsed to %temp% my path had become C:\Users(my-username-here)\AppData\Local\Temp\4.

I have never run into this issue with Windows Server 2016, so I am thinking maybe the problem is related to some kind of Windows Server 2019 bug. I'll post back here for posterity if I discover anything else.
